doc/versioning_and_life_cycle.md

```diff
@@ -0,0 +1,73 @@
+
+### Relationship with api-version
+
+At present, CLI sets the api-version according to the Resource Provider granularity, and there are corresponding configurations for different profiles. For most commands, we do not support customers to specify the requested api-version.
+
+- For the GA api-version, if the command itself determines the final design, then we usually make the implemented command also in the GA state.
+
+- For the public preview api-version, the situation is relatively complicated, and we usually divide it into the following cases:
+    1. if those features are newly onboard RPs, command groups or extensions, we usually mark them as `is_preview`. 
+    2. If there are some existing commands under these RPs, but only the new features need to be released through the preview api-version, and we need to add some new commands or parameters to support them, then we need to upragde to the preview api-version and only mark `is_preview` on those new commands and parameters, while the existing commands and parameters remain in the GA state. We still need to ensure that the existing commands and parameters are stable without breaking changes.
+    3. Due to the higher flexibility of CLI extension, we want customers to use the stable extension version by default, so we plan to make the release state of extension consistent with the api-version state: if the api-version is preview, the whole extension should be in preview state. And we need to add "preview" suffix in extension version number *(comply with [pep-0440](https://peps.python.org/pep-0440/))* to prevent customers from upgrading to preview version automatically. We will add support for `az extension add` and `az upgrade` to allow customers to install/upgrade to stable version as default behavior and to install/upgrade to preview version by specifying extra parameter in the future. 
+
+- For the private preview api-version, we usually do not recommend release those private features in core CLI/CLI extension, so as to avoid premature exposure of private features. It is suggested to use [edge build](https://github.com/Azure/azure-cli/blob/dev/doc/try_new_features_before_release.md) to build the private packages to meet the urgent needs of some customers first.
+
+Since the release states of api-version and CLI commands may be inconsistent, we hope that customers only need to pay attention to the release states *(`is_preview` flag)* of the command itself, rather than the api-version used behind it, which will be easier for customers to understand and use it.
```

Comment:
```suggestion
Since the release states of api-version and CLI commands may be inconsistent, our intent is for customers to only have to pay attention to the release states *(`is_preview` flag)* of the command itself, rather than the api-version used behind it.  This will be easier for customers to understand.
```